exonum_client/api.py

```diff
@@ -0,0 +1,552 @@
+              "state": {
+                "delay": 4000,
+                "type": "Reconnect"
+              }
+            }
+          }
+        }
+
+        Returns
+        -------
+        response: requests.Response
+            Result of an API call.
+            If it is successful, a list of peers will be returned.
+        """
+        return self.get(self._system_url.format("peers"))
+
+    def set_consensus_interaction(self, enabled: bool = True) -> requests.Response:
+        """
+        Performs a POST request to the '{system_base_path}/consensus_enabled'
+        to switch consensus interaction of the node on or off.
+
+        Parameters
+        ----------
+        enabled: flag that switches consensus interaction of the node on (True) or off (False).
+
+        Returns
+        -------
+        response: requests.Response
+            Result of an API call.
+        """
+        data = json.dumps({"enabled": enabled})
+        return self.post(
+            self._system_url.format("consensus_enabled"), data=data, headers={"content-type": "application/json"}
+        )
+
+    def get_consensus_interaction(self) -> requests.Response:
+        """
+        Performs a GET request to the '{system_base_path}/consensus_enabled' to get boolean value that states
+        if the node participates in consensus.
+
+        Example:
+
+        >>> private_api = PrivateApi("127.0.0.1", 81)
+        >>> private_api.get_consensus_interaction().json()
+        True
+
+        Returns
+        -------
+        response: requests.Response
+            Result of an API call.
+        """
+        return self.get(self._system_url.format("consensus_enabled"))
+
+    def get_network_info(self) -> requests.Response:
+        """
+        Performs a GET request to the '{system_base_path}/network'
+        to get info about the serialization protocol and the services functioning in the network.
+
+        Example:
+
+        >>> private_api = PrivateApi("127.0.0.1", 81)
+        >>> network_info = private_api.get_network_info().json()
+        >>> print(json.dumps(network_info, indent=2))
+        {
+          "core_version": "0.10.2"
+        }
+
+        Returns
+        -------
+        response: requests.Response
+            Result of an API call.
+        """
+        return self.get(self._system_url.format("network"))
+
+    def shutdown(self) -> requests.Response:
+        """
+        Performs a POST request to the '{system_base_path}/shutdown' to stop the node.
+        After receiving this request, the node stops processing transactions, participating in consensus and
+        terminates after all messages in the event queue are processed.
+
+        Returns
+        -------
+        response: requests.Response
+            Result of an API call.
+        """
+        data = json.dumps(None)
+        return self.post(self._system_url.format("shutdown"), data=data, headers={"content-type": "application/json"})
+
+
+class ServiceApi:
```

Comment:
I think you can simplify this: use only one port and provide two methods in `ExonumClient`: `service_private_api(service_name: str)` and ``service_public_api(service_name: str)`.

Then it will be used in the code like:
```python
supervisor_public_api = client.service_private_api("supervisor")
supervisor_private_api = client.service_private_api("supervisor")
supervisor_private_api.post(...)
data = supervisor_public_api.get(...)
````

Evolving this idea, you can add a helper method `service_apis(service_name: str)` which will return a tuple of both public and private api:

```python
public_api, private_api = client.service_apis("supervisor")
```

Reply:
Actually I was thinking about such approach, but then decided to stick to jira description.